src/common/components/recovery-account/index.tsx

```diff
@@ -23,33 +23,59 @@ import {
 } from "../../api/operations";
 
 import "./index.scss";
+import {
+  addRecoveries,
+  deleteRecoveries,
+  getRecoveries,
+  getRecoveriesEmail
+} from "../../api/private-api";
+import { FullAccount } from "../../store/accounts/types";
 
 interface Props {
   global: Global;
   activeUser: ActiveUser | null;
   signingKey: string;
   setSigningKey: (key: string) => void;
 }
+
 export default function AccountRecovery(props: Props) {
   const [keyDialog, setKeyDialog] = useState(false);
   const [inProgress, setInProgress] = useState(false);
   const [disabled, setDisabled] = useState(true);
+  const [isEcency, setIsEcency] = useState(false);
   const [popOver, setPopOver] = useState(false);
   const [step, setStep] = useState(1);
   const [toError, setToError] = useState("");
+  const [accountData, setAccountData] = useState<FullAccount>();
+  const [recoveryEmail, setRecoveryEmail] = useState("");
+  const [toWarning, setToWarning] = useState("");
+  const [recoveryEmails, setRecoveryEmails] = useState<getRecoveriesEmail[]>([]);
   const [currRecoveryAccount, setCurrRecoveryAccount] = useState("");
   const [newRecoveryAccount, setNewCurrRecoveryAccount] = useState("");
   const [pendingRecoveryAccount, setPendingRecoveryAccount] = useState("");
 
+  const ECENCY = "ecency";
```

Comment:
Please, move it outside of component declaration because this property will be created each render time